src/core/core.controller.js

```diff
@@ -225,9 +225,6 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 			me.resize(true);
 		}
 
-		// Make sure scales have IDs and are built before we build any controllers.
-		me.ensureScalesHaveIDs();
-		me.buildOrUpdateScales();
 		me.initToolTip();
```

Comment:
Are we sure that `initToolTip()` and `plugins.notify(me, 'afterInit');` don't expect scales to be correctly initialized?

Reply:
`InitTooltip()` does not, but who knows what there is in `afterInit`

Reply:
So it needs to be decided if we support the expectation of scales being initialized in `afterInit`.
I guess a plugin might cache scales there, but then it would not work if those scales are changed later. So any plugin caching scales there would be slightly broken actually 🙊 
I can't think of any other use cases (scales don't have any dimensions etc here, so not really usable).

So opinions needed here, @etimberg @nagix @simonbrunel @benmccann

Reply:
I think it is ok to remove them. As @kurkle says, if a plugin needs to do something using scales, it is natural to implement logic in `afterUpgrade`. No config is available in `afterInit` either.

Reply:
The only plugin I know of that uses scales is https://github.com/bbc/Chart.Bands.js/blob/master/src/chart.bands.js